hnn_core/network_builder.py

```diff
@@ -425,35 +437,184 @@ def _create_cells_and_feeds(self):
             else:
                 _PC.cell(gid, feed_cell.nrn_netcon)
 
+    def _connect_celltypes(self, src_type, target_type, loc,
+                           receptor, nc_dict, unique=False,
+                           allow_autapses=True):
+        """Connect two cell types for a particular receptor.
+
+        Parameters
+        ----------
+        src_type : str
+            The source cell type
+        target_type : str
+            The target cell type
+        loc : str
+            If 'proximal' or 'distal', the corresponding
+            dendritic sections from Cell.sect_loc['proximal']
+            or Cell.Sect_loc['distal'] are used
+        receptor : str
+            The receptor.
+        nc_dict : dict
+            The connection dictionary containing keys
+            A_delay, A_weight, lamtha, and threshold.
+        unique : bool
+            Is unique?
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        """
+        net = self.net
+        connection_name = f'{src_type}_{target_type}_{receptor}'
+        if connection_name not in self.ncs:
+            self.ncs[connection_name] = list()
+        for gid_target in net.gid_dict[_long_name(target_type)]:
+            if _PC.gid_exists(gid_target):
+                gid_srcs = net.gid_dict[_long_name(src_type)]
+                if unique:
+                    gid_srcs = [gid_target + net.gid_dict[src_type][0]]
+                for gid_src in gid_srcs:
```

Comment:
For a unique feed, it looks almost looks like each source gid is getting connected to *all* cells of the given type. Is that correct?

Reply:
nopes, for a unique feed, `gid_srcs` is a list containing *one* element. So you don't have a nested for loop. Instead you loop only over the target cells and `gid_src` is actually "inferred" from the `gid_target` rather than being read from `net.gid_dict` ... so yeah agree it is hackish. What I did is just a reorganization of the old logic. I think fixing this might be a great idea for another PR.

Reply:
I get it now. Thanks for the explanation!